torchvision/utils.py

```diff
@@ -375,13 +376,16 @@ def _generate_color_palette(num_masks: int):
     return [tuple((i * palette) % 255) for i in range(num_masks)]
 
 
-@no_type_check
-def _log_api_usage_once(obj: str) -> None:  # type: ignore
-    if torch.jit.is_scripting() or torch.jit.is_tracing():
-        return
-    # NOTE: obj can be an object as well, but mocking it here to be
-    # only a string to appease torchscript
-    if isinstance(obj, str):
-        torch._C._log_api_usage_once(obj)
-    else:
-        torch._C._log_api_usage_once(f"{obj.__module__}.{obj.__class__.__name__}")
+def _log_api_usage_once(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        event = f.__module__
+        if f.__name__.endswith("__init__"):
+            # inside module instantiation
+            event += args[0].__class__.__name__
```

Comment:
I think we'll need a dot in the middle to preserve the previous naming (same below)

```suggestion
            event += "." + args[0].__class__.__name__
```

Reply:
Good catch, I addressed in local, but haven't pushed here yet.